cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/resource/StatefulSetOperator.java

```diff
@@ -114,45 +95,39 @@ public StatefulSetOperator(Vertx vertx, KubernetesClient client, long operationT
             return Future.succeededFuture();
         } else {
             Future<Void> result = Future.future();
-            log.info("Rolling update of {}/{}: Rolling pod {}", namespace, name, podName);
-            Future<Void> deleted = Future.future();
             Future<CompositeFuture> deleteFinished = Future.future();
-            Watcher<Pod> watcher = new RollingUpdateWatcher(deleted);
-            Watch watch = podOperations.watch(namespace, podName, watcher);
+            log.info("Rolling update of {}/{}: Rolling pod {}", namespace, name, podName);
+
+            // Determine generation of deleted pod
+            Future<Integer> deleted = getGen(namespace, podName);
+
             // Delete the pod
-            log.debug("Rolling update of {}/{}: Waiting for pod {} to be deleted", namespace, name, podName);
-            Future podReconcileFuture = podOperations.reconcile(namespace, podName, null);
-            CompositeFuture.join(podReconcileFuture, deleted).setHandler(deleteResult -> {
-                watch.close();
+            Future<ReconcileResult<Pod>> podReconcileFuture = deleted.compose(l -> {
+                log.debug("Rolling update of {}/{}: Waiting for pod {} to be deleted", namespace, name, podName);
+                return podOperations.reconcile(namespace, podName, null);
```

Comment:
This seems a bit confusing. I was quite wondering how do you delete the pod. Would it be better to add a comment or call `internalDelete` directly?